Assignments/ip1/ip1.md

```diff
@@ -0,0 +1,328 @@
+Students implemented a single game, [Tic-Tac-Toe](https://en.wikipedia.org/wiki/Tic-tac-toe), as a proof-of-concept for this feature.
+
+The objective for this semester's individual project is to extend this new `GameArea` abstraction, implementing [Connect Four](https://en.wikipedia.org/wiki/Connect_Four) as a new game that can be played within a `GameArea`.
+This implementation effort will be split across two deliverables. In this first deliverable, you will implement and test the core backend components for this feature, and in the second deliverable, you will implement and test the frontend components. 
+
+## Objectives of this assignment
+The objectives of this assignment are to:
+*  Get you familiar with the basics of TypeScript, VSCode, and the project codebase
+*  Learn how to read and write code in TypeScript
+*  Translate high-level requirements into code
+*  Learn how to write unit tests with Jest
+
+## Getting started with this assignment
+
+Before you begin, be sure to check that you have NodeJS 18.x installed, along with VSCode. We have provided a [tutorial on setting up a development environment for this class]({{site.baseurl}}% link tutorials/week1-getting-started.md % **FIX LINK**) 
+Start by [downloading the starter code]({{site.baseurl}}{% link /Assignments/ip1/ip1-handout.zip %}). Extract the archive and run `npm install` to fetch the dependencies. Avery has provided you with some very basic sanity tests that you can extend for testing your implementation as you go. You can run these tests with the command `npm test ConnectFour` (note that many tests are *expected* to fail until you have begun to implement the assignment).
+
+## Grading
+This submission will be scored out of 100 points, 90 of which will be automatically awarded by the grading script, with the remaining 10 manually awarded by the course staff.
+
+Your code will automatically be evaluated for linter errors and warnings. Submissions that have *any* linter errors will automatically receive a grade of 0. **Do not wait to run the linter until the last minute**. To check for linter errors, run the command `npm run lint` from the terminal. The handout contains the same eslint configuration that is used by our grading script.
+
+Your code will be automatically evaluated for functional correctness by a test suite that expands on the core tests that are distributed in the handout. 
+Your tests will be automatically evaluated for functional correctness by a process that will inject bugs into our reference solution: to receive full marks your tests must detect a minimum number of injected bugs. 
+Each submission will be graded against the same set of injected bugs (repeated submissions will not receive new/different injected bugs).
+You will __not__ receive detailed feedback on which injected bugs you do or do not find.
```

Comment:
TODO: Add info on hints, decide what the hints per submission and per day should be.